harness/determined/launch/deepspeed.py

```diff
@@ -236,6 +221,21 @@ def main(script: List[str]) -> int:
         if dtrain_network_interface is not None and len(dtrain_network_interface) > 0:
             os.environ["NCCL_SOCKET_IFNAME"] = dtrain_network_interface
 
+    # In some downstream training code, the hostfile is expected on all nodes so we create the
+    # hostfile on all nodes here before opening the non-chief node subprocess.  Since we
+    # use the allocation id to create the hostfile path, we register that path to an environment
+    # variable for access downstream.  A filelock is used to ensure we do not have clashing writes
+    # if the hostfile_path is on a shared filesystem.
+    hostfile_path = get_hostfile_path(multi_machine, info.allocation_id)
+
+    lock = str(hostfile_path) + ".lock"
+    with filelock.FileLock(lock):
```

Comment:
filelocks don't get deleted if a process dies. so in theory this could result in a bunch of "temporary" lock files on the shared fs. not sure if this is a big problem, but something to note.

Reply:
i think this is okay, usually hostfiles are persistent anyway and ours would get cleaned up most of the time with the container unless /tmp is bind mounted